src/classes/ADDR_Addresses_UTIL.cls

```diff
@@ -0,0 +1,618 @@
+    * @return Boolean
+    */ 
+    public static string getMultilineStreet(Address__c addr) {
+        string str = addr.MailingStreet__c;
+        if (addr.MailingStreet2__c != null && addr.MailingStreet2__c.length() > 0)
+            str += '\r\n' + addr.MailingStreet2__c;
+        return str;
+    }
+    
+    /*******************************************************************************************************
+    * @description specifies whether state and country picklists are enabled in this org.
+    * @return true if enabled
+    */ 
+    public static Boolean isStateCountryPicklistsEnabled {
+        get {
+            if (isStateCountryPicklistsEnabled == null)
+                isStateCountryPicklistsEnabled = initValidStateCountryCodes();
+            return isStateCountryPicklistsEnabled;
+        }
+        private set;
+    }
+    
+    /*******************************************************************************************************
+    * @description For the list of Addresses, sees if there is already an existing Address in the Account that
+    * matches. Creates only unique addresses from their respective household, and mark any existing matches 
+    * as Default if the caller specifies it on the new address.
+    * @param listAddr a list of Addresses to update
+    * @param dmlWrapper the Addresses to update
+    * @param includeAddressType whether to include comparing the AddressType field
+    * @return void
+    */ 
+    public static List<Address__c> createNonDuplicateAddresses(list<Address__c> listAddr, TDTM_Runnable.DmlWrapper dmlWrapper, 
+    Boolean includeAddressType) {  
+        
+        List<Address__c> addrsToInsert = new List<Address__c>();
+        map<Address__c, Address__c> mapAddrDupAddr= findDuplicateAddresses(listAddr);
+        for (Address__c addrNew : listAddr) {
+            Address__c addrDup = mapAddrDupAddr.get(addrNew);
+            if (addrDup != null) {
+                // if it is a fuzzy match, not exact, then we want to update the existing address.
+                if (isAddressChanged(addrNew, addrDup, includeAddressType)) {
+                    copyAddressObj(addrNew, addrDup);
+                    if (addrNew.Default_Address__c) {
+                        addrDup.Default_Address__c = true;
+                    }
+                    dmlWrapper.objectsToUpdate.add(addrDup);
+                } else if (addrNew.Default_Address__c && !addrDup.Default_Address__c) {
+                    addrDup.Default_Address__c = true;
+                    dmlWrapper.objectsToUpdate.add(addrDup);
+               }
+            } else { 
+                addrsToInsert.add(addrNew);
+            } 
+        }
+        //Inserting them directly because we need the IDs to populate the Current_Address__c field in Account
+        if(addrsToInsert.size() > 0) {
+            UTIL_Debug.debug('****addrsToInsert: ' + JSON.serializePretty(addrsToInsert));
+            insert addrsToInsert;
```

Comment:
This DML statement is not surrounded by any try-catch block. Please add a try-catch block to ensure exceptions are handled properly.